.github/workflows/linkcheck.yml

```diff
@@ -27,18 +29,22 @@ jobs:
     - name: Checkout code
       uses: actions/checkout@v4
       with:
-        fetch-depth: 0
+        persist-credentials: false
 
-    - name: Set up Python
-      uses: actions/setup-python@v5
-      with:
-        python-version: 3.12
+    - name: Install Graphviz
+      uses: ts-graphviz/setup-graphviz@v2
+
+    - name: Install pandoc
+      # This step requires Ubuntu
+      run: |
+        sudo apt update
+        sudo apt-get install -y pandoc
 
-    - name: Install Python dependencies
-      run: python -m pip install --upgrade nox uv
+    - name: Install uv
+      uses: astral-sh/setup-uv@v5
 
     - name: Install graphviz & pandoc
       run: sudo apt-get install graphviz pandoc
 
     - name: Check hyperlinks
-      run: nox -s linkcheck
+      run: uv run --python 3.12 --with nox --frozen nox -s linkcheck
```

Comment:
The `--frozen` is so that `uv.lock` doesn't get updated during `uv run`.  Without it, `uv.lock` would get updated and the runs would no longer be reproducible.